src/components/structureDefSettings/utils.ts

```diff
@@ -2,31 +2,45 @@ import {
   IElementDefinition_Type,
   IStructureDefinition
 } from "@ahryman40k/ts-fhir-types/lib/R4";
-import { PrimitiveTypesType } from "state/actions/fhirDataTypesActions";
 import { RenderAttributesTree } from "types";
 
 export const editAttribute = <K extends keyof IStructureDefinition>(
   structureToEdit: IStructureDefinition,
   attributeKey: K,
   attributeValue?: IStructureDefinition[K]
 ) => {
-  if (undefined !== attributeValue && attributeValue.toString() !== "") {
+  if (attributeValue !== undefined && attributeValue.toString() !== "") {
     structureToEdit[attributeKey] = attributeValue;
     return structureToEdit;
   } else {
     delete structureToEdit[attributeKey];
   }
 };
 
-export const createJSONTree = (items: RenderAttributesTree[]) => {
+/**
+ * Create a JSON tree from a structure definition
+ * @param items
+ * A tree of type RenderAttributesTree that will help to create a JSON tree as in FHIR resources
+ * @param base
+ * Original structure definition of the resource we choosed to profile
+ */
+export const createJSONTree = (items: RenderAttributesTree[], base: any) => {
   const sDef: any = {};
   for (const item of items) {
     if (item.children.length === 0) {
       sDef[item.name] = undefined;
     } else {
       if (item.name !== "snapshot" && item.name !== "differential") {
-        if (item.max === "1") sDef[item.name] = createJSONTree(item.children);
-        else sDef[item.name] = [createJSONTree(item.children)];
+        if (item.max === "1")
```

Comment:
It is difficult to read, there are lots of imbricated if 🙂

my improvement proposition (to be tested):
```
  items.forEach((item: RenderAttributesTree) => {
    if (item.children.length === 0) {
      sDef[item.name] = undefined;
    } else if (item.name !== "snapshot" && item.name !== "differential") {
      if (item.max === "1") {
        sDef[item.name] = createJSONTree(item.children, base);
      } else {
        sDef[item.name] = [];
        if (base[item.name]) {
          base[item.name].forEach(() => sDef[item.name].push(createJSONTree(item.children, base)));
        }
      }
    }
  })
```

I think it could be cool to type sDef 🙂